src/vs/vscode.d.ts

```diff
@@ -4325,6 +4330,29 @@ declare module 'vscode' {
 		resolveTask(task: Task, token?: CancellationToken): ProviderResult<Task>;
 	}
 
+	export enum LogLevel {
+		Trace = 1,
+		Debug = 2,
+		Info = 3,
+		Warning = 4,
+		Error = 5,
+		Critical = 6,
+		Off = 7
+	}
+
+	export interface ILogger {
+		onDidChangeLogLevel: Event<LogLevel>;
```

Comment:
https://github.com/Microsoft/vscode/wiki/Extension-API-guidelines#global-events

Reply:
Actually depends... When this is in the `env` it should be surely global when it stays in the extension context maybe not... Needs thinking

Reply:
@mjbvz are imagining different log levels for different things. I think there's a good argument for being able to enable tracing for just one extension. So I think the logLevel and change event should be on the extension's own logger.